tools/applicationinsights-web-snippet/src/snippet.ts

```diff
@@ -202,11 +204,60 @@ declare var cfg:ISnippetConfig;
             "js2.cdn.applicationinsights.io",
             "js2.cdn.monitor.azure.com",
             "az416426.vo.msecnd.net" // this domain is supported but not recommended
-        ]
-    
-        // Assigning these to local variables allows them to be minified to save space:
-        let targetSrc : string = (aiConfig as any)["url"] || cfg.src
-        if (targetSrc) {
+        ];
+
+        let targetSrc : string = (aiConfig as any)["url"] || cfg.src;
+        var match = targetSrc.match(/^(http[s]?:\/\/.*\/)([\w]+\.)?(\d+(\.\d+){0,2})\.(([\w]+\.){0,2}js)$/);
+        if (match.length === 7 && cfg.sri) {
+            var integrityUrl = match[1] + "ai." + match[3] + ".integrity.json";
+            var targetType = "@" + match[5];
+            var sender = window.fetch;
+            var integrity: string = null;
+            if (sender && !cfg.useXhr) {
+                // retrieve integrity file using fetch
+                sender(integrityUrl, { method: strGetMethod, mode: "cors" })
+                    .then(response => response.json().catch(() => ({}))) // return an empty object, will be catched by the next if statement
+                    .then(json => {
+                        if (!json.ext || !json.ext[targetType]) {
+                            throw new Error(strJsonResponseError);
+                        }
+                        integrity = json.ext[targetType].integrity;
+                        targetSrc = match[1] + json.ext[targetType].file;
+                        setScript(targetSrc, integrity);
+                    })
+                    .catch(error => {
+                        console.error(strJsonResponseError, error);
+                        setScript(targetSrc, null); // Fallback to original behavior
+                    });
+            } else if (XMLHttpRequest) {
+                var xhr = new XMLHttpRequest();
+                xhr.open(strGetMethod, integrityUrl);
+                xhr.onreadystatechange = function () {
+                    if (xhr.readyState === XMLHttpRequest.DONE && xhr.status === 200) {
+                        try {
+                            var json = JSON.parse(xhr.responseText);
+                            if (!json.ext || !json.ext[targetType]) {
+                                throw new Error(strJsonResponseError);
+                            }
+                            integrity = json.ext[targetType].integrity;
+                            targetSrc = match[1] + json.ext[targetType].file;
```

Comment:
We probably should also check if `integrity` and `targetSrc` contain values (as someone could have a json file that doesn't include these) and if we pass null/undefined onto the `setScript` it won't be happy either and will throw on the first line.